src/Adapters/Cache/RedisCacheAdapter.js

```diff
@@ -33,9 +35,15 @@ export class RedisCacheAdapter {
     debug('put', key, value, ttl);
     this.p = this.p.then(() => {
       return new Promise((resolve, _) => {
-        this.client.set(key, value, function(err, res) {
-          resolve();
-        });
+        if (ttl) {
```

Comment:
The logic in InMemoryCache is a little different:
TTL = 0 means store but have immediately expire, e.g. in tests.
TTL = NaN means never expire.

Your change has TTL = 0 never expiring and I'm not sure what happens when NaN is passed to psetex().
I would change to match InMemoryCache